include/evmc/evmc.h

```diff
@@ -502,36 +502,114 @@ typedef evmc_bytes32 (*evmc_get_storage_fn)(struct evmc_host_context* context,
  * notation is used:
  * - 0 is zero value,
  * - X != 0 (X is any value other than 0),
- * - Y != X, Y != 0 (Y is any value other than X and 0),
- * - Z != Y (Z is any value other than Y),
- * - the "->" means the change from one value to another.
+ * - Y != 0, Y != X,  (Y is any value other than X and 0),
+ * - Z != 0, Z != X, Z != X (Z is any value other than Y and X and 0),
+ * - the "o -> c -> v" triple describes the change status in the context of:
+ *   - o: original value (cold value before a transaction started),
+ *   - c: current storage value,
+ *   - v: new storage value to be set.
+ *
+ * The order of elements follows EIPs introducing net storage gas costs:
+ * - EIP-2200: https://eips.ethereum.org/EIPS/eip-2200,
+ * - EIP-1283: https://eips.ethereum.org/EIPS/eip-1283.
  */
 enum evmc_storage_status
 {
     /**
-     * The value of a storage item has been left unchanged: 0 -> 0 and X -> X.
+     * The value of a storage item has been left unchanged or has been modified again
+     * in the transaction context. This is the group of "no-op" cases related to
+     * minimal gas cost.
+     * 0|X   -> 0 -> 0 (current value unchanged)
+     * 0|X|Y -> Y -> Y (current value unchanged)
+     * 0|X   -> Y -> Z (modified previously added/modified value)
+     *
+     * This is "catch all remaining" status. I.e. if all other statuses are correctly matched
+     * this status should be assigned to all remaining cases.
+     *
+     * This corresponds to the EIP-2200 rule 2 or rule 3.2 without sub-rules, where:
+     * c == v or (c != v, o != c, (o == 0 or (o != 0, c != 0, v != 0)), o != v)
```

Comment:
This is hard to parse, and rule numbers like `3.2.1.1` are not useful because don't exist in EIP.

I perfectly understand notation like `X -> Y -> 0`, why do you need this new one?

Reply:
This is redundant. I try to explain what EIP rules it matches with the ordered list on condition to reach the particular rules in the spec tree in the EIP.

Any ideas how to improve this?

Reply:
Do you think it is better to remove it? I can have separate doxygen chapter (md document) that describes all matching details using big tables.